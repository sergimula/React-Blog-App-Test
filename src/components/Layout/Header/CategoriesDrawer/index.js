import React, { useState } from 'react'
import { Drawer, Grid, IconButton, makeStyles } from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'
import PropTypes from 'prop-types'

import Logo from 'src/components/Layout/Logo'
import CategoriesList from './CategoriesList'
import styles from './styles'

const useStyles = makeStyles(styles)

const CategoriesDrawer = (props) => {
  const [isDrawerOpen, setIsDrawerOpen] = useState(false)
  const classes = useStyles()

  const toggleDrawer = (isDrawerOpen) => (event) => {
    if (
      event.type === 'keydown' &&
      (event.key === 'Tab' || event.key === 'Shift')
    ) {
      return
    }

    setIsDrawerOpen(isDrawerOpen)
  }

  return (
    <React.Fragment>
      <Grid
        alignContent='center'
        alignItems='center'
        container
        justify='center'
      >
        <Grid className={classes.drawerMenuIcon} item>
          <IconButton onClick={toggleDrawer(true)}>
            <MenuIcon style={{ color: props.burgerMenuIconColor }} />
          </IconButton>
        </Grid>
        <Grid item>
          <Logo />
        </Grid>
      </Grid>
      <Drawer anchor={'left'} open={isDrawerOpen} onClose={toggleDrawer(false)}>
        <CategoriesList
          categories={props.categories}
          changeCategoryByID={props.changeCategoryByID}
          toggleDrawer={toggleDrawer}
        />
      </Drawer>
    </React.Fragment>
  )
}

CategoriesDrawer.propTypes = {
  burgerMenuIconColor: PropTypes.string,
  categories: PropTypes.array,
  changeCategoryByID: PropTypes.func,
}

export default CategoriesDrawer
