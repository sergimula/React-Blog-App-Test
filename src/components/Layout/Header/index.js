import React from 'react'
import CategoriesTabs from '../../navigation/CategoriesTabs'
import PropTypes from 'prop-types'

const Header = (props) => {
  if (!props.categories) return null

  return (
    <CategoriesTabs
      activeTabId={props.activeCategoryID}
      burgerMenuIconColor={props.burgerMenuIconColor}
      changeCategoryByID={props.changeCategoryByID}
      categories={[{ id: 0, name: 'Home' }, ...props.categories]}
      headerBackgroundColor={props.headerBackgroundColor}
    />
  )
}

Header.propTypes = {
  activeCategoryID: PropTypes.number,
  burgerMenuIconColor: PropTypes.string,
  categories: PropTypes.array,
  changeCategoryByID: PropTypes.func,
  headerBackgroundColor: PropTypes.string,
}

export default Header
