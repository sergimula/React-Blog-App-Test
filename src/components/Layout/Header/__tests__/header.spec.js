import React from 'react'
import renderer from 'react-test-renderer'
import Tabs from '@material-ui/core/Tabs'

import Header from '../index'
import * as stubs from './stubs'

describe('Header', () => {
  it('should match Header', () => {
    const wrapper = renderer.create(<Header {...stubs.headerProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('when categories are empty', () => {
    it('should not render the Header', () => {
      const wrapper = renderer.create(
        <Header {...stubs.headerPropsCategoriesEmpty} />
      )

      expect(wrapper.toJSON()).toBe(null)
    })
  })
})
