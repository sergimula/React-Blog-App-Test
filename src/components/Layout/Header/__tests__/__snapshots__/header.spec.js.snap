// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match Header 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorDefault mui-fixed MuiPaper-elevation4"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root makeStyles-drawerMenuIcon-3 MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "color": "",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <Logo />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <Tabs here
        className="makeStyles-tabs-2"
        indicatorColor="primary"
        onChange={[Function]}
        textColor="primary"
        value={0}
        variant="scrollable"
      >
        <button
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Home
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Coding
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Motorcycles
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Entertainment
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            News
          </span>
        </button>
      </Tabs here>
    </div>
  </header>
</div>
`;
